frontends/web/src/routes/exchange/btcdirect.tsx

```diff
@@ -51,13 +51,9 @@ export const BTCDirect = ({ accounts, code }: TProps) => {
   const { isDarkMode } = useDarkmode();
 
   const iframeRef = useRef<HTMLIFrameElement | null>(null);
-  const btcdirectInfo = useLoad(() => getBTCDirectInfo('buy'));
+  const btcdirectInfo = useLoad(() => getBTCDirectInfo('buy', code));
 
-  const receiveAddresses = useLoad(getReceiveAddressList(code));
-
-  // TODO: address should probably come from the backend, i.e. ETH address
-  const p2wpkhAddresses = receiveAddresses?.find(({ scriptType }) => scriptType === 'p2wpkh')?.addresses || [];
-  const address = p2wpkhAddresses[0]?.address || '';
+  const address = btcdirectInfo?.success ? btcdirectInfo.address : '';
```

Comment:
Not super super sure how to handle the case in which success is false here, though it is also handled below (line 123)

Reply:
It will only load the iframe if address has a "truthy" value (see below `{ address && `), empty string is falsy.

Truthy/falsy is often a bit weird to read. You could change from empty string to undefined then it might be more clear.

Reply:
I don't think you need to define the `address` const at all. There is a check for `btcdirectInfo?.success` just before including the iframe, so you can just call `btcdirectInfo.address` there, as done for the apiKey and the url.

Reply:
Good point, done!